### DISCUSSION TITLE
respect query string in html conversion

### DISCUSSION THREAD

**Discussion by stefanke** [gotenberg/gotenberg#1007] [Q&A] [answered]
We would like to use Gotenberg together with a NocoDB button to create PDF files. We need to add some custom arguments through the NocoDB fields as a string query.

This works fine when we create a preview button in NocoDB. The button arguments inside NocoDB are, for example:
CONCAT("https://internalwebsitename.net/", {Template}, "/?project=", {Project_Title}, "&document_title=", {Document_Subtitle}, "&version_number=", {Version}, "&versionDate=", {Date})
The {}-fields are handled by NocoDB, so we end up with a standard HTML string query.

However, when I send this string to Gotenberg, it doesn't work. The PDF is created, but the extra arguments are ignored.

Is there a way to pass the string query arguments to Gotenberg so that it renders our HTML site correctly with the extra queries?

Thank you for your help or any hints,
Stefan

**Comment by gulien** [comment, marked as answer]
Hello @stefanke,

Gotenberg should take the whole URL without truncating it in any way. Are you certain the URL is correctly sent to Gotenberg via the `url` form field? Could you share the debug log of your Gotenberg container? 🤔

**Reply by stefanke** [reply]
Hi Gulien,

you are right, gotenberg does not truncate the URL at all. It was my javscript helper-script. Everything works as expected. 

Thanks a lot, problem solved!
